src/components/Input.tsx

```diff
@@ -1,129 +1,116 @@
 import { Feather } from '@expo/vector-icons'
-import { forwardRef, useCallback, useState } from 'react'
-import { ChangeHandler, Control, Controller, get, Path, RegisterOptions } from 'react-hook-form'
-import { FieldErrors } from 'react-hook-form/dist/types/errors'
-import { StyleProp, StyleSheet, TextInput, TextInputProps, View, ViewStyle } from 'react-native'
+import { IInputProps, FormControl, useTheme, Input as NBInput } from 'native-base'
+import { forwardRef } from 'react'
+import {
+  ChangeHandler,
+  Control,
+  Controller,
+  FieldErrors,
+  Path,
+  RegisterOptions,
+  get,
+} from 'react-hook-form'
+import { TextInput } from 'react-native'
 
-import { Box, Touchable } from './Atoms'
-import { Text } from './Typography'
+import { Touchable } from './Atoms'
 
-import { useTheme, useMemo, useSecurePassword } from '~hooks'
+import { useSecurePassword, useCallback, useMemo } from '~hooks'
 
-type InputProps = TextInputProps & {
-  placeholder: string
+type InputProps = IInputProps & {
   label?: string
-  error?: boolean
-  inputContainerStyle?: StyleProp<ViewStyle>
+  helperText?: string
+  errorMessage?: string
+  errorIcon?: JSX.Element
   onInputBlur?: () => ChangeHandler
-  fullWidth?: boolean
-  type?: InputTypes
 }
 
+const layoutPropsKeys = [
```

Comment:
Is it really necessary with native base?

Reply:
In the way, I've built that input - yes - layout propses would be passed to both FormControl and Input.